### DISCUSSION TITLE
Off Axis in Bevy and Lighting/Shadows

### DISCUSSION THREAD

**Discussion by nicholasbl** [bevyengine/bevy#15861] [Q&A]
I'm exploring using Bevy to build a multi-process immersive (cave-like) renderer to potentially replace a home-grown C++ library. The design of Bevy has made this a pretty fun process, and I'm enjoying the flexibility.

One stumbling block I'm hitting is on custom projection. While there doesn't appear to be many up-to-date docs on the topic, I believe I have the basics working with my custom off-axis projection type and CameraProjectionPlugin + PbrProjectionPlugin. However, it seems that for some of the more extreme-angled screens we are working with, shadows are not working (i.e. it looks like the depth buffer is corrupted-- for the most part, everything is testing as being in shadow, with a few screen-aligned blocks flickering, but only for those certain screens). 

This is pretty niche, so I don't want to take up folks' time, but I was wondering if there is another plugin I might have missed that I should be adding in before I start digging into the shaders.

Thanks!

**Comment by JMS55** [comment]
I wouldn't be surprised to find out that several of our shaders/pieces of rendering code assume perspective/ortho projections, or worse, perspective only. We don't tend to test on custom projections ever, and rendering algorithms often takes shortcuts for perspective projections.
